src/xbot/common/controls/sensors/MockXboxController.java

```diff
@@ -90,42 +94,24 @@ public void setRightTriggerPressed(double x){
         rightTriggerAxis = x;
     }
     
-    public XYPair getLeftStick(double x, double y){
-        return leftJoystickAxis = new XYPair(x, y);
-    }
-    
-    @Override
-    public XYPair getLeftStick() {
-        return null;
-    }
-    
-    public XYPair getRightStick(double x, double y){
-        return rightJoystickAxis = new XYPair(x, y);
-    }
-    
-    @Override
-    public XYPair getRightStick() {
-        return null;
-    }
-    
     @Override
     public double getLeftStickX() {
-        return leftJoystickAxis.x;
+        return getLeftVector().x;
     }
     
     @Override
     public double getRightStickX() {
-        return rightJoystickAxis.x;
+        return getRightVector().x;
     }
 
     @Override
     public double getLeftStickY() {
-        return leftJoystickAxis.y;
+        return getLeftVector().y;
     }
 
     @Override
     public double getRightStickY() {
-        return rightJoystickAxis.y;
+        return getRightVector().y;
```

Comment:
Invert this, so that the vector getter calls the  individual ones. Every call to `getRightVector` creates a vector object, yet here it's immediately discarded.